disk/datadog_checks/disk/disk.py

```diff
@@ -357,3 +364,20 @@ def _get_devices_label(self):
             self.log.debug("Couldn't use blkid to have device labels")
 
         return devices_label
+
+    def _get_devices_label_from_blkid_cache(self):
+        devices_label = {}
+        try:
+            with open(self._blkid_cache_file, 'r') as blkid_cache_file_handler:
+                # Line sample
+                # <device DEVNO="0x0801" LABEL="MYLABEL" UUID="..." TYPE="ext4">/dev/sda1</device>
+                for line in blkid_cache_file_handler:
+                    device = self._blkid_device_re.findall(line)
+                    label = self._blkid_label_re.findall(line)
+                    if label and device:
+                        devices_label[device[0]] = 'label:{}'.format(label[0])
```

Comment:
It's preferable to not keep the file open during the parsing and reduce the scope of the try/except.

```python
try:
    with open() as file:
        blkid_cache_data = file.read()
except:
    ...

# parsing and extraction logic outside of the try/except
```

Reply:
Done